test/pthread/test_pthread_lsan_no_leak.cpp

```diff
@@ -13,7 +13,6 @@ void g(void) {
   void *stuff = malloc(3432);
   tls_ptr = malloc(1234);
   atomic_store(&thread_done, true);
-  printf("thread done\n");
```

Comment:
How is it possible that this might not show up before "LSAN TEST COMPLETE"?   That main thread verifies that thread_done is set, which can only happen once "thread done\n" has been printed..

Is the problem perhaps that one is stderr and the other is stdout?

Reply:
I mean, the atomic operation is before this print, and after it, the threads race to see who can print first.

Reply:
Oh I see.. then the fix is probably just to move the print to before the atomic store.  `printf` should not return until the main thread has fully buffered the string to be printed.  

Any print operations on the main thread that come after the atomic store would then be guaranteed to come after this print.

Reply:
It could be moved, but it seemed simpler to remove it to me. It serves no purpose and the test is shorter without it?

Reply:
If you are sure it not needed then that sounds fine to me.